Use best practices for wandb.init() and the Run object (#1146)
wandb/docs: content/guides/integrations/add-wandb-to-any-library.md

Before you get started, decide whether or not to require W&B in your library’s dependencies:

> #### Require W&B on installation

Reviewer: This and the next heading seem odd to me. It seems like you need to choose whether to require it, and depending on your choice, you need to take an action either way. What if these were collapsed into a single section "Add W&B as a dependency" and start with "If you add it to your dependencies file, it is required. To make it optional, <foo>.

Second, in the next section, I'd change the A and B to list items, or sub headings (but we are getting deep). You'll need to indent your code blocks two stops, I believe.

Initialize W&B and start a Run within your code with:

```python
run = wandb.init()
```

Optionally, you can provide a name for their project, or let the user set it themselves with parameters such as `wandb_project` in your code along with the username or team name, such as `wandb_entity`, for the entity parameter:

```python
run = wandb.init(project=wandb_project, entity=wandb_entity)
```

> It is important to eventually call `run.finish()` on the run. If this works with your integartion's design, the most convenient way to do this is by using the run as a context manager:

Reviewer: ```suggestion
You must call `run.finish()` to finish the run. If this works with your integration's design,  use the run as a context manager:
```
Typo and copyediting

Author: Oof, thanks for catching that!

```python
# When this block exits, it calls run.finish() automatically.
# If it exits due to an exception, it uses run.finish(exit_code=1) which
# marks the run as failed.
with wandb.init() as run:
    ...
```

> #### Where To Place `wandb.init`?

Reviewer: This is "when to call", not "where to place". right?

Author: Updated! I also updated other headings in this file to use sentence case.

    "val/loss": 0.5, 
    "val/accuracy": 0.7
}
run.log(metrics)
```

{{< img src="/images/integrations/integrations_add_any_lib_log.png" alt="A W&B Workspace with 2 separate sections" >}}

> For more on `run.log`, see [here]({{< relref "/guides/models/track/log" >}}).

Reviewer: ```suggestion
[Learn more about `run.log`]({{< relref "/guides/models/track/log" >}}).
```

#### Prevent x-axis misalignments

> Sometimes you might need to perform multiple calls to `run.log` for the same training step. The wandb SDK has its own internal step counter that is incremented every time a `run.log` call is made. This means that there is a possibility that the wandb log counter is not aligned with the training step in your training loop.

Reviewer: ```suggestion
If you perform multiple calls to `run.log` for the same training step, the wandb SDK increments an internal step counter for each call to `run.log`. This counter may not align with the training step in your training loop.
```

> To avoid this, we recommend that you specifically define your x-axis step. You can define the x-axis with `run.define_metric` and you only need to do this once, after `wandb.init` is called:

Reviewer: ```suggestion
To avoid this situation, define your x-axis step explicitly with `run.define_metric`, one time, immediately after you call `wandb.init`:
```

```python
with wandb.init(...) as run:
    run.define_metric("*", step_metric="global_step")
```

The glob pattern, `*`, means that every metric will use `global_step` as the x-axis in your charts. If you only want certain metrics to be logged against `global_step`, you can specify them instead:

```python
run.define_metric("train/loss", step_metric="global_step")
```

> Now that you've called `run.define_metric`, you just need to log your metrics as well as your `step_metric`, `global_step`, every time you call `run.log`:

Reviewer: ```suggestion
Now, log your metrics, your `step` metric, and your `global_step` each time you call `run.log`:
```

  * For images, you can log sample predictions, segmentation masks, etc., to see the evolution over time.
  * For text, you can log tables of sample predictions for later exploration.

> Refer [here]({{< relref "/guides/models/track/log" >}}) for a full guide on logging media, objects, plots, and more.

Reviewer: ```suggestion
[Learn more about logging]({{< relref "/guides/models/track/log" >}}) media, objects, plots, and more.
```
